hi-ml-histopathology/src/histopathology/datasets/base_dataset.py

```diff
@@ -73,19 +75,28 @@ def __init__(self,
             self.dataset_csv = dataset_csv or self.root_dir / self.DEFAULT_CSV_FILENAME
             dataset_df = pd.read_csv(self.dataset_csv)
 
-        columns = [self.SLIDE_ID_COLUMN, self.IMAGE_COLUMN, self.LABEL_COLUMN,
-                   self.SPLIT_COLUMN, self.TILE_X_COLUMN, self.TILE_Y_COLUMN]
-        for column in columns:
-            if column is not None and column not in dataset_df.columns:
-                raise ValueError(f"Expected column '{column}' not found in the dataframe")
-
         dataset_df = dataset_df.set_index(self.TILE_ID_COLUMN)
         if train is None:
             self.dataset_df = dataset_df
         else:
             split = self.TRAIN_SPLIT_LABEL if train else self.TEST_SPLIT_LABEL
             self.dataset_df = dataset_df[dataset_df[self.SPLIT_COLUMN] == split]
 
+        if validate_columns:
+            self.validate_columns()
+
+    def validate_columns(self) -> None:
+        """Check that loaded dataframe contains expected columns, raises `ValueError` otherwise.
+
+        If the constructor is overloaded in a subclass, you can pass `validate_columns=False` and
+        call `validate_columns()` after creating derived columns, for example.
+        """
+        columns = [self.SLIDE_ID_COLUMN, self.IMAGE_COLUMN, self.LABEL_COLUMN,
+                   self.SPLIT_COLUMN, self.TILE_X_COLUMN, self.TILE_Y_COLUMN]
+        for column in columns:
+            if column is not None and column not in self.dataset_df.columns:
+                raise ValueError(f"Expected column '{column}' not found in the dataframe")
```

Comment:
Maybe printing the found columns would also be helpful for debugging. E.g., "... not found in data frame with columns '{self.dataset_df.columns}'"

Reply:
If we really want a helpful error message, it should collect _all_ missing columns before raising the error. Otherwise you fix one column, re-run, see another failure.

Reply:
It is currently consistent for both `TilesDataset` and `SlidesDataset`